cypress/integration/chi-vue/search-input.spec.js

```diff
@@ -0,0 +1,207 @@
+  });
+
+  describe('Base', () => {
+    it('Should exist base search input', () => {
+      cy.get(`${SEARCH_INPUTS_DATA_CY.BASE}`)
+    });
+  });
+
+  describe('Disabled', () => {
+    it('Should disable search input', () => {
+      cy.get(`${SEARCH_INPUTS_DATA_CY.DISABLED}`)
+        .find(`${ELEMENTS.INPUT}`)
+        .should('have.attr', 'disabled');
+    });
+  });
+
+  describe('Sizes', () => {
+    const sizes = ['sm', 'md', 'lg', 'xl'];
+
+    sizes.forEach((size) => {
+      it(`Should have the class -${size} size`, () => {
+        checkSizeClass(`[data-cy='${size}-search-input-test']`, `-${size}`);
+      });
+    });
+  });
+
+  describe('Attrs', () => {
+    const attrs = ['name', 'placeholder'];
+    const booleanAttrs = ['portal', 'prevent-value-mutation']
+
+    attrs.forEach((attr) => {
+      it(`Should have ${attr} attr`, () => {
+        checkAttr(`${SEARCH_INPUTS_DATA_CY.ATTRS}`, `${ELEMENTS.INPUT}`, attr, 'test');
+      });
+    });
+
+    booleanAttrs.forEach((attr) => {
+      it(`Should have the ${attr} attr set to true`, () => {
+        if (attr === 'portal') {
+          checkPortalAttr(`${SEARCH_INPUTS_DATA_CY.ATTRS}`, `${COMPONENT_CLASSES.PORTAL}`)
+          return;
+        }
+
+        checkBooleanAttr(`${SEARCH_INPUTS_DATA_CY.ATTRS}`, attr, 'true');
+      });
+    });
+  });
+
+  describe('Events', () => {
+    it(`Should trigger chiBlur`, () => {
+      cy.window()
+        .its(`${EVENT_EXAMPLES.BLUR}`)
+        .then(chiBlur => {
+          const component = chiBlur.$refs.testEvents;
+          const spy = cy.spy();
+
+          component.$on(`${EVENTS.BLUR}`, spy);
+          cy.get(`${SEARCH_INPUTS_DATA_CY.EVENTS}`).find(`${ELEMENTS.INPUT}`).focus().blur()
+            .then(() => {
+              expect(spy).to.be.called;
+            });
+        });
+    });
+
+    it(`Should trigger chiChange`, () => {
+      cy.window()
+        .its(`${EVENT_EXAMPLES.CHANGE}`)
+        .then(chiChange => {
+          const component = chiChange.$refs.testEvents;
+          const spy = cy.spy();
+          const value = 'test change';
+
+          component.$on(`${EVENTS.CHANGE}`, spy);
+          cy.get(`${SEARCH_INPUTS_DATA_CY.EVENTS}`).find(`${ELEMENTS.INPUT}`).type(value).blur()
+            .then(() => {
+              checkValue(`${SEARCH_INPUTS_DATA_CY.EVENTS}`, value);
+              expect(spy).to.be.called;
+            });
+        });
+    });
+
+    it(`Should trigger chiClean`, () => {
+      cy.window()
+        .its(`${EVENT_EXAMPLES.CLEAN}`)
+        .then(chiClean => {
+          const component = chiClean.$refs.testEvents;
+          const spy = cy.spy();
+
+          component.$on(`${EVENTS.CLEAN}`, spy);
+          cy.get(`${SEARCH_INPUTS_DATA_CY.EVENTS}`).find(`${ELEMENTS.INPUT}`).type('test')
+          cy.get(`${SEARCH_INPUTS_DATA_CY.EVENTS}`).find(`.${COMPONENT_CLASSES.CLOSE}`).click()
+            .then(() => {
+              expect(spy).to.be.called;
+            });
```

Comment:
Here we are checking if the event has been emitted, but we are not checking if the value has been set to an empty array (basically, the clean functionality).

Reply:
Solved.